extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzScheduler.java

```diff
@@ -155,7 +177,18 @@ public QuartzScheduler(SchedulerContext context, QuartzSupport quartzSupport, Co
                                 scheduled);
                     }
                 }
-            } catch (SchedulerException e) {
+                if (nonManagedTx) {
+                    if (handle.isAvailable()) {
+                        try {
+                            handle.get().commit();
+                        } catch (HeuristicMixedException | HeuristicRollbackException | RollbackException e) {
+                            handle.get().rollback();
```

Comment:
I believe that we should always rollback if something goes wrong, ie. right before `throw new IllegalStateException("Unable to create Scheduler", e)`. Actually, we could change the catch block below to `catch Throwable`...